### ISSUE TITLE
Sprint 141 Planning - OSIO Analytics

### ISSUE THREAD

**Issue by krishnapaparaju** [openshiftio/openshift.io#1403]
### Sprint Goals 
**Work towards publishing IDE extensions for OSIO analytics platform (VSCode, DevStudio)**
- [x] Test and increase analytics coverage for Vertx, SpringBoot (#1287)
- [ ] UX related enhancements for the stack report  (#1277) 
- [ ] Identify performance bottlenecks and publish matrix for OSIO resource requirements based on load testing results (#1144)
- [ ] Deploy 3scale APICast gateway, Update IDE workflow to include 3scale user registration and test IDE workflow end-to-end with 3Scale gateway deployed. 
- [ ] List and close all the marketplace related TODOs (icons, READMEs) for publishing IDE extensions

**Analytics recommendations for quick-starter app** 
- [ ] https://github.com/openshiftio/openshift.io/issues/1471

**Plan and design Machine generated stacks covering user experience**
- [ ] Complete POC for Machine generated stacks covering user experience (DEMO)
        - [ ] UI (https://github.com/openshiftio/openshift.io/issues/1323)
        - [ ] Backend (https://github.com/openshiftio/openshift.io/issues/1462)
- [ ] Work closely with Corrine and document current components and changes to components for the front end / UI side of wizard/launcher ..

**Wrap up significant % of ingestion for the GO ecosystem (#939)**
- [ ] Continue ingesting Go packages: #939
- [x] Bug in Go support: #1394
- [ ] Glide support in mercator: #1406 

**Figure out next steps for operationalising CVE mapping work to individual packages  (#1264)**
- [ ] #1260

### Bugs
- [x] #1430
- [ ] #1429
- [x] #1428
- [x] #1427
- [ ] #1450
- [x] #1449
- [x] #1288
- [x] #787
- [x] ~~#1433~~ caused by rh-idev cluster being overloaded, nothing that we can fix on our side
- [ ] #1298
- [x] #1027 
- [x] #768 
- [ ] #1147 
- [x] #1407 
- [x] https://github.com/openshiftio/openshift.io/issues/1483
- [x] https://github.com/openshiftio/openshift.io/issues/1489
- [x] https://github.com/openshiftio/openshift.io/issues/1491
- [x] https://github.com/openshiftio/openshift.io/issues/1485
- [x] https://github.com/openshiftio/openshift.io/issues/1486

### Engineering work

**Continue the progress on developers using Dev. cluster and wrap up remaining TODOs**
- [x] finish: #1062
- [ ] Make it possible to easily run E2E tests against f8a deployments on the dev cluster: #1258 

**Work on high priority Technical Debt**
- [ ] Configure SQS once per deployment: #1398
- [x] UTC timestamps: #1359
- [ ] Expose workers bookkeeping data: #1404
- [x] Technical Debt: Integration Test for stack dependency topic list validation #1458

**Wrap up service related open items for getting a handle on various operational scenarios, service (#1087)**

##### AWS resources/infra improvements:
- [ ] Map AWS resources to staging/dev #1081

##### Graph sync, spike this sprint:
- [ ] #1085

##### Tests:
- [ ] enable/improve E2E tests for ingestion components: #1329

##### Gradle:
- [ ] Implement Gradle support in Mercator: #689

### Stretch Goals

- [ ] Wrap up Crowdsourcing (inside, outside stack report) for VSCode and DevStudio extensions

- [ ] Fix identified performance bottlenecks
     
- [ ] Calculate the % of packages that could not tagged across NPM (#1095)

- [ ] Setup end to end flow for users, NPM Recommendations would not be accurate by sprint 141

- [ ] Move all fabric8-analytics images to "fabric8-analytics" namespace: #1196

- [ ] Dockerize deployment scritps: #1405

- [ ] Infra Issue #1142

- [ ] #688
 
- [ ] Delete unnecessary AWS resources if required for operational needs: #1330 (depends on #1081)

**Comment by msrb** [comment]
Tasks proposed for the core stream:

Go:
- continue ingesting Go packages: https://github.com/openshiftio/openshift.io/issues/939
- bug in Go support: https://github.com/openshiftio/openshift.io/issues/1394
- glide support in mercator: https://github.com/openshiftio/openshift.io/issues/1406

Dev cluster workflow:
- finish: https://github.com/openshiftio/openshift.io/issues/1062
- dockerize deployment scritps: https://github.com/openshiftio/openshift.io/issues/1405
- move deploy scripts to a separate repo: https://github.com/openshiftio/openshift.io/issues/1383

AWS resources/infra improvements:
- map aws resources to staging/dev, ...: https://github.com/openshiftio/openshift.io/issues/1081
- delete unnecessary AWS resources if required for operational needs: https://github.com/openshiftio/openshift.io/issues/1330


CVEs next steps:
- https://github.com/openshiftio/openshift.io/issues/1260

Graph sync:
- https://github.com/openshiftio/openshift.io/issues/1085

Gradle:
- implement Gradle support in Mercator: https://github.com/openshiftio/openshift.io/issues/689

Tests:
- enable/improve E2E tests for ingestion components: https://github.com/openshiftio/openshift.io/issues/1329


Bugs/improvements:
- configure sqs once per deployment: https://github.com/openshiftio/openshift.io/issues/1398
- utc timestamps: https://github.com/openshiftio/openshift.io/issues/1359
- expose workers bookkeeping data: https://github.com/openshiftio/openshift.io/issues/1404

**Comment by tisnik** [comment]
Tasks for the QA team (on member :):

1. enable/improve E2E tests for ingestion components: #1329
2. utc timestamps: #1359 
3. Configure the fabric8 server module to use Swagger-UI and make the API Swagger-compliant https://github.com/openshiftio/openshift.io/issues/1327

**Comment by samuzzal-choudhury** [comment]
Marketplace related TODOs identified by developers are -

#### VSCode

1. Need to have readme's verified as this will be shown in marketplace [1] [2]
2. Get the icon for extensions
3. Need to finalise on Extension names as well(We have 2 extensions for VSCode)
4. Need to automate the flow to publish the extension to marketplace as redhat official
5. 

Developer Studio:
NA

**Comment by krishnapaparaju** [comment]
@samuzzal-choudhury @msrb @miteshvp @sivaavkd @GeorgeActon  

These stream level planned activities would need to be grouped into few work items (for the entire platform) as this consolidated list of work items would need to reviewed by PM team...

**Comment by krishnapaparaju** [comment]
@tisnik require QE help out on performance, user facing SLA. (load testing with different combinations of resources) and sharing results with the development team...

**Comment by samuzzal-choudhury** [comment]
@krishnapaparaju @GeorgeActon @sivaavkd @miteshvp @msrb - Please review the following sprint deliverables for the entire team and suggest any modifications.


#### Sprint Goal 
- Publish IDE extensions for OSIO analytics platform (VSCode, DevStudio)

    - [ ] Test and increase analytics coverage for Vertx, SpringBoot (#1287)

    - [ ] UX related enhancements for the stack report  (#1277) 

    - [ ] Identify performance bottlenecks and publish matrix for OSIO resource requirements based on load testing results (#1144)

    - [ ] Deploy 3scale APICast gateway, Update IDE workflow to include 3scale user registration and test IDE workflow end-to-end with 3Scale gateway deployed. 

    - [ ] List and close all the marketplace related TODOs (icons, READMEs) for publishing IDE extensions

- Plan and design Machine generated stacks covering user experience

    - [ ] Complete POC for Machine generated stacks covering user experience (https://github.com/openshiftio/openshift.io/issues/1323)

    - [ ] Work closely with Corrine and document current components and changes to components for the front end / UI side of wizard/launcher ..

- Wrap up significant % of ingestion for the GO ecosystem (#939)

    - [ ] Continue ingesting Go packages: #939

    - [ ] Bug in Go support: #1394

    - [ ] Glide support in mercator: #1406

- Figure out next steps for operationalising CVE mapping work to individual packages  (#1264)
    - [ ] #1260

#### Engineering work: 

- Continue the progress on developers using Dev. cluster and wrap up remaining TODOs

    - [ ] finish: #1062

    - [ ] move deploy scripts to a separate repo: #1383

- Work on high priority Technical Debt.

- Continue the progress on developers using Dev. cluster and wrap up remaining TODOs

    - [ ] finish: #1062

    - [ ] move deploy scripts to a separate repo: #1383

- Wrap up service related open items for getting a handle on various operational scenarios, service (#1087)

    - AWS resources/infra improvements:

        - [ ] Map AWS resources to staging/dev #1081

        - [ ] Delete unnecessary AWS resources if required for operational needs: #1330

    - Graph sync:

        - [ ] #1085

    - Tests:

        - [ ] enable/improve E2E tests for ingestion components: #1329


- Gradle:

    - [ ] Implement Gradle support in Mercator: #689


- Bugs/improvements:

    - [ ] Configure SQS once per deployment: #1398

    - [ ] UTC timestamps: #1359

    - [ ] Expose workers bookkeeping data: #1404

#### Stretch Goals

- [ ] Wrap up Crowdsourcing (inside, outside stack report) for VSCode and DevStudio extensions

- [ ] Fix identified performance bottlenecks
     
- [ ] Calculate the % of packages that could not tagged across NPM (#1095)

- [ ] Setup end to end flow for users, NPM Recommendations would not be accurate by sprint 141

- [ ] Move all fabric8-analytics images to "fabric8-analytics" namespace: #1196

- [ ] Dockerize deployment scritps: #1405

- [ ] Infra Issue #1142

- [ ] #688

**Comment by bmicklea** [comment]
Wow, that's a huge sprint. Do you expect to complete all the items listed above stretch?  Are they in priority order?

**Comment by krishnapaparaju** [comment]
Its a huge sprint. Couple of sprint goals are spill overs from last train. Work items under 'Sprint goals', 'Bugs' carry higher priority than work items at 'Engineering work'

**Comment by bmicklea** [comment]
FWIW operationalizing the CVE information so that it can appear in analytics reports is a high business priority.

**Comment by krishnapaparaju** [comment]
@bmicklea Ack.

**Comment by GeorgeActon** [comment]
Sprint 141 is complete and open items have been moved to Sprint 142.
